tools/WindowStylesReportTool/WindowStylesReportTool.cpp

```diff
@@ -0,0 +1,424 @@
+#include "pch.h"
+#include "WindowStylesReportTool.h"
+
+#include <dwmapi.h>
+#include <shlobj.h>
+
+#include <filesystem>
+#include <fstream>
+#include <map>
+
+inline std::optional<std::wstring> get_last_error_message(const DWORD dw)
+{
+    std::optional<std::wstring> message;
+    try
+    {
+        const auto msg = std::system_category().message(dw);
+        message.emplace(begin(msg), end(msg));
+    }
+    catch (...)
+    {
+    }
+    return message;
+}
+
+inline std::wstring get_last_error_or_default(const DWORD dw)
+{
+    auto message = get_last_error_message(dw);
+    return message.has_value() ? message.value() : L"";
+}
+
+std::filesystem::path get_desktop_path()
+{
+    wchar_t* p;
+    if (S_OK != SHGetKnownFolderPath(FOLDERID_Desktop, 0, NULL, &p)) return "";
+
+    std::filesystem::path result = p;
+    CoTaskMemFree(p);
+
+    return result;
+}
+
+// Get the executable path or module name for modern apps
+inline std::wstring get_process_path(DWORD pid) noexcept
+{
+    auto process = OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION | PROCESS_VM_READ, TRUE, pid);
```

Comment:
```suggestion
    auto process = OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, TRUE, pid);
```

Small misunderstanding. 😅 Changing like suggested and then it works great.

Reply:
Hi @htcfreek.
Could you please explain this change?

Reply:
Asking because I'm just waiting for CI to get done to merge this PR in so we can build some release candidates.

Reply:
this is related to a change before to get the name from elevated windows without running elevated itself.

@SeraphimaZykova only change `Query_Process` to `Query_Process_Limmited` and missed to rome ``

Reply:
I have tried the change on my local build. Should be save to merge.

Reply:
So, if I was targetting a elevated process the current code would fail, that's what you're telling me, right?

Reply:
I'll give it a test in a bit.

Reply:
It's the same we do in WW plugin. (See #15186.)

Not sure if it fail. But log has empty string for window process.

Reply:
Just tested it. This fixes it, indeed. Thank you!